.builds/openbsd.yml

```diff
@@ -1,34 +1,30 @@
-image: openbsd/6.9
+image: openbsd/7.0
 
 packages:
-  - gmake
-  - automake-1.16.3
-  - autoconf-2.71
-  - openssl-1.1.1k
+  - meson
+  - pkgconf
+  - openssl-3.0.0
   - lzo2
   - lz4
   - miniupnpc
   - readline
   - texinfo
 
-environment:
-  AUTOCONF_VERSION: 2.71
-  AUTOMAKE_VERSION: 1.16
-  CFLAGS: -I/usr/local/include -L/usr/local/lib
-
 sources:
   - https://github.com/gsliepen/tinc
 
+environment:
+  LD_LIBRARY_PATH: /usr/local/lib/eopenssl30
```

Comment:
I just tried to build it on OpenBSD 7.0 myself, and I was surprised it didn't detect OpenSSL or LibreSSL support automatically, while this should just be part of the default install. It turns out it is there, and even has a pkg-config file, but OpenBSD's pkg-config file says it is openssl version 1.0.0.

It would be great if we do support OpenBSD with the default LibreSSL install, maybe a compromise is to detect if we are on OpenBSD, and then lower the minimum required version of OpenSSL to 1.0.0.

Reply:
LibreSSL was forked from OpenSSL 1.0.x and its primary goal was to:
- remove support for all systems other than OpenBSD, since cross-platform support leads to code complexity
- remove legacy, deprecated ciphers rather than leaving them in as deprecated

Actually supporting OpenSSL 1.1.x APIs, which projects increasingly started using and depending on, was promised for years, while nothing ever really happened. And of course OpenBSD is not willing to accept Apache 2.0 code, so they cannot import changes from OpenSSL.

Anyway if you do not *require* (on an API level) openssl >=1.1.0 you don't need to version guard based on it. :)

Reply:
LibreSSL seems to have support for all the APIs the project needs (unlike 'real' OpenSSL 1.0).

---

Empty fallback [doesn't work][ci] with meson 0.55, so I had to write this gem.

Hopefully we can start using the meta-dependency in a couple of years and the pain goes away.

[ci]: https://github.com/hg/tinc/runs/5650645312?check_suite_focus=true#step:7:173